### PR TITLE
Drop `UserHistories` provider

### PR DESCRIPTION
**Pull request by itisAliRH**
Follow up on #15108

- Dropping the `UserHistories` provider and updating all files using it
- Improve some components to use history store actions directly instead of passing them down

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: client/src/stores/historyStore.ts [hunk 1]
```diff
@@ -149,7 +149,8 @@ export const useHistoryStore = defineStore(
             const deletedHistory = (await deleteHistoryById(historyId, purge)) as HistorySummary;
             Vue.delete(storedHistories.value, deletedHistory.id);
             if (getFirstHistoryId.value) {
-                return setCurrentHistoryId(getFirstHistoryId.value);
+                setCurrentHistoryId(getFirstHistoryId.value);
+                return setCurrentHistory(getFirstHistoryId.value);
```
Comment (ElectronicBlueberry) [inline client/src/stores/historyStore.ts]: This seems strange, as `setCurrentHistory` also calls `setCurrentHistoryId` via `selectHistory`, so it is now being called twice.

What is the reason for this change, and could it also be done within `setCurrentHistory` or `selectHistory`?

File: client/src/stores/historyStore.ts [hunk 2]
```diff
@@ -145,14 +145,21 @@ export const useHistoryStore = defineStore(
...
 
+        function getNextAvailableHistoryId(excludedIds: string[]) {
+            const historyIds = Object.keys(storedHistories.value);
+            const filteredHistoryIds = historyIds.filter((id) => !excludedIds.includes(id));
+            return filteredHistoryIds[0];
+        }
+
         async function deleteHistory(historyId: string, purge: boolean) {
             const deletedHistory = (await deleteHistoryById(historyId, purge)) as HistorySummary;
-            Vue.delete(storedHistories.value, deletedHistory.id);
```
Comment (itisAliRH) [inline client/src/stores/historyStore.ts:157]: @ElectronicBlueberry The problem was that when history was deleted, the `currentHistory` became undefined and an error was raised until it was replaced in the following.

File: client/src/components/Workflow/Run/WorkflowRunForm.vue [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <template>
-    <UserHistories v-if="currentUser" v-slot="{ currentHistoryId }" :user="currentUser">
+    <div v-if="currentUser">
```
Comment (ElectronicBlueberry) [inline client/src/components/Workflow/Run/WorkflowRunForm.vue]: This div can be dropped

File: client/src/components/Tour/Tour.vue [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <template>
     <div class="d-flex flex-column">
-        <UserHistories v-if="currentUser" v-slot="{ currentHistory, handlers, historiesLoading }" :user="currentUser">
+        <div v-if="currentUser">
```
Comment (ElectronicBlueberry) [inline client/src/components/Tour/Tour.vue]: This divs condition can be attached to the parent to reduce nesting.
The condition may not even be required any longer now that the provider is dropped. I could not find any change in tour behavior when dropping it completely (without reattaching the condition)

File: client/src/components/Tool/ToolForm.vue [hunk 5]
```diff
@@ -1,6 +1,6 @@
 <template>
     <ConfigProvider v-slot="{ config }">
-        <UserHistories v-if="currentUser" v-slot="{ currentHistoryId }" :user="currentUser">
+        <div v-if="currentUser">
```
Comment (ElectronicBlueberry) [inline client/src/components/Tool/ToolForm.vue]: This condition can be attached to it's child div to reduce nesting

File: client/src/components/Markdown/Elements/HistoryDatasetCollection/CollectionDisplay.vue [hunk 6]
```diff
@@ -13,7 +13,7 @@
                     class="py-0 px-1">
                     <span class="fa fa-download" />
                 </b-button>
-                <UserHistories v-if="user" v-slot="{ currentHistoryId }" :user="currentUser">
+                <div v-if="currentUser">
```
Comment (ElectronicBlueberry) [inline client/src/components/Markdown/Elements/HistoryDatasetCollection/CollectionDisplay.vue]: This divs condition can be attached to the button directly, if it is even still required

File: client/src/components/History/Multiple/MultipleView.vue [hunk 7]
```diff
@@ -16,48 +16,40 @@ const filter = ref("");
...
+                <DebouncedInput v-slot="{ value, input }" v-model="filter">
+                    <b-form-input
+                        size="sm"
+                        :class="filter && 'font-weight-bold'"
+                        :value="value"
+                        :placeholder="'search datasets in selected histories' | l"
+                        data-description="filter text input"
+                        @input="input"
+                        @keyup.esc="updateFilter('')" />
+                </DebouncedInput>
```
Comment (ElectronicBlueberry) [inline client/src/components/History/Multiple/MultipleView.vue]:
```suggestion
                <b-form-input
                    size="sm"
                    v-model="filter"
                    debounce="500"
                    :class="filter && 'font-weight-bold'"
                    :placeholder="localize('search datasets in selected histories')"
                    data-description="filter text input"
                    @keyup.esc="updateFilter('')" />
```

Fixes the typescript errors

File: client/src/components/Markdown/Elements/HistoryDatasetCollection/CollectionDisplay.vue (hunk 6 above)
Comment (itisAliRH) [reply client/src/components/Markdown/Elements/HistoryDatasetCollection/CollectionDisplay.vue]: I applied the change but I can't approve it while the collection display is not working #16048

**Review by davelopez — APPROVED** [review]
Looks good to me! Thank you @itisAliRH!

**Review by ElectronicBlueberry — APPROVED** [review]
